### PR TITLE
All products

### PR DESCRIPTION
**Pull request by LevinCa**
Created API GET-Request for all products + tests and implemented frontend with single page `/menu`

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -10,9 +10,13 @@
     "@types/node": "^16.18.22",
     "@types/react": "^18.0.31",
     "@types/react-dom": "^18.0.11",
+    "axios": "^1.3.4",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^6.10.0",
     "react-scripts": "5.0.1",
+    "react-toastify": "^9.1.2",
+    "toastify": "^2.0.1",
```
Comment (Flooooooooooorian) [inline frontend/package.json:19]: remove unused librarys

File: frontend/src/ProductContext.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const [allProduct, setAllProducts] = useState<Product[]>([])
+
+    useEffect(() =>
+        {
+            getAllProducts()
+        }, []
+    )
+
+    function getAllProducts(): void {
+        axios.get("api/product")
```
Comment (Flooooooooooorian) [inline frontend/src/ProductContext.tsx:18]:
```suggestion
        axios.get("/api/product")
```

File: frontend/src/product/card/ProductCard.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    product: Product
+}
+
+export default function ProductCard(props: ProductCardProps) {
+
+
+    return (
+        <div className={"ProductContainer"}>
+            <div className={"ProductCard"}>
+                <img src={props.product.imageURL} alt={"Image not found"}/>
```
Comment (Flooooooooooorian) [inline frontend/src/product/card/ProductCard.tsx:14]: Find a better alternative

File: frontend/src/product/gallery/ProductGallery.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+
+export default function ProductGallery() {
+
+    const context = useContext(ProductProvider)
+
+    return (
+        <div className={"ProductGallery"}>
+            {context.allProducts.map(p => {
+                return <ProductCard key={p.id} product={p}/>
```
Comment (Flooooooooooorian) [inline frontend/src/product/gallery/ProductGallery.tsx:14]:
```suggestion
            {context.allProducts.map(product => {
                return <ProductCard key={product.id} product={product}/>
```

File: frontend/src/static/NavigationBar.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import "./NavigationBar.css"
+
+
+export default function NavigationBar() {
+    return (
+        <div className={"NavigationBar"}>
+            <NavLink className={"NavigationItem"} to={"/home"}>Home</NavLink>
+            <NavLink className={"NavigationItem"} to={"/menu"}>Menü</NavLink>
+            <NavLink className={"NavigationItem"} to={"/order"}>Bestellen</NavLink>
+            <NavLink className={"NavigationItem"} to={"/about"}>Kontakt</NavLink>
```
Comment (Flooooooooooorian) [inline frontend/src/static/NavigationBar.tsx:11]: Remove not existing routes

**Review by Flooooooooooorian — APPROVED** [review]

File: frontend/src/static/NavigationBar.tsx (hunk 5 above)
Comment (LevinCa) [reply frontend/src/static/NavigationBar.tsx:11]: nope

**Pull request merged**